File: server/python/app/presentation/models/hosted_checkout/request.py
from pydantic import BaseModel, field_validator, ConfigDict
from typing import Optional
from pydantic.alias_generators import to_camel
from app.application.domain.enums.common.currency import Currency
from app.application.dtos.common.address_dto import AddressDto

class HostedCheckoutRequest(BaseModel):
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    amount: float
    currency: Currency
    language: Optional[str] = None
    redirect_url: Optional[str] = None
    billing_address: Optional[AddressDto] = None
    shipping_address: Optional[AddressDto] = None

    @field_validator('amount', mode='before')
    @classmethod
    def validate_amount(cls, v):
        if v is None:
            return v
        if isinstance(v, str) and v.strip() == "":
            return None
        try:
            return float(v)
        except (ValueError, TypeError):
            raise ValueError("The Amount field must be a number.")

File: server/ruby/app/presentation/models/create_hosted_checkout/request.rb
module Presentation
  module Models
    module CreateHostedCheckout
      class Request
        attr_accessor :amount, :currency, :language, :redirect_url, :billing_address, :shipping_address

        def initialize(amount: nil, currency: nil, language: nil, redirect_url: nil, billing_address: nil, shipping_address: nil)
          @amount           = amount
          @currency         = currency
          @language         = language
          @redirect_url     = redirect_url
          @billing_address  = billing_address
          @shipping_address = shipping_address
        end
      end
    end
  end
end